### ISSUE TITLE
How to run CenterHMR on CPU?

### ISSUE THREAD

**Issue by JANVI2411** [Arthur151/CenterHMR#20]
First of all thanks @Arthur151. I'm able to generate 3D meshes for my videos but I'm curious to know if we can also run this on CPU. it would be very kind of you if you can guide me on this.

**Comment by Arthur151** [comment]
Thanks for your interests! 
I have just commit a version that support CPU. 
You just have to change the `GPU: 0` to `GPU: -1` in config files (like src/configs/basic_test.yml or src/configs/basic_test_video.yml). Then the code should be run on CPU.

For details, please check the last two commits:
https://github.com/Arthur151/CenterHMR/commit/12f46d12d52e0174247016543ff4debed7e597fc
https://github.com/Arthur151/CenterHMR/commit/4004a0d7f4cd1e3aed2dfe54efcc4d1ef53acea9

**Comment by JANVI2411** [comment]
Thanks for the quick reply. 
I tried changing GPUS: 0 to GPUS:-1 in src/configs/basic_test.yml but unfortunately it's throwing an error "TypeError: argument of type 'int' is not iterable"  in this below line: 

if '-1' not in self.gpu  (file: src/core/base.py , line no: 41,55,189)

I changed this line to " if self.gpu!=-1 " then the above error is gone but got this error: "RuntimeError: No CUDA GPUs are available"

Sharing with you a screenshot : https://drive.google.com/file/d/1v9kjrOdI8nXWgjSIK21Zpp-qjUzrRYcm/view?usp=sharing

**Comment by Arthur151** [comment]
There are several things need to be noticed.
1. I have just upload the commit. So you have to 'git pull' to drag the update. Or please simply reload the Colab and start over the entire process.
2. There is a space between 'GPUS:' and '-1'. `GPUS: -1` not `GPUS:-1` .

**Comment by JANVI2411** [comment]
Yes. I already have taken care of both of these things. I setup the repository in totally new colab file and also put space between GPUS: and -1 (You can see that in the above screenshot , on right side I've kept open the yml file for your reference ). 

why did you put this if condition in only base.py file as cuda() is used in other files too.

**Comment by Arthur151** [comment]
Sorry for the bug. I have just try the Colab. It turns out that the pyrender need GPU for visualization. So current code could run on CPU and save the 3D body mesh / SMPL parameter files. But it won't render the estimated 3D meshes back to 2D image due to the limitation of Pyrender package. I will fix this bug tomorrow. In the future, I will try to replace the visualization module to achieve full CPU support.

**Comment by Arthur151** [comment]
Sorry for the bug. I have just try the Colab. It turns out that the pyrender need GPU for visualization. So current code could run on CPU and save the 3D body mesh / SMPL parameter files. But it won't render the estimated 3D meshes back to 2D image due to the limitation of Pyrender package. I will fix this bug tomorrow. In the future, I will try to replace the visualization module to achieve full CPU support.

**Comment by Arthur151** [comment]
Don't hesitate to let me know your problem. 
Best.

**Comment by JANVI2411** [comment]
Sure !
Thanks for your time and guidance.
Please keep the thread updated if anything pops up.

**Comment by Arthur151** [comment]
We have relase a new version with more funny things. Hope you enjoy it.

**Comment by Arthur151** [comment]
Training code has been released along with all evaluation.

**Comment by Arthur151** [comment]
@JANVI2411 
BEV has been released for training, inference, and evaluation.
